<launch>
 <arg name="scene_file_arg" default="/home/cgoodin/mavs/data/scenes/forester_scene_less_trees_simple.json" /> 
  <arg name="vehicle_file_arg" default="/home/cgoodin/mavs/data/vehicles/rp3d_vehicles/mrzr4_tires.json" />

  <arg name="cmd_vel_topic_arg" default="avt_341/cmd_vel" />
  <arg name="point_cloud_topic_arg" default="avt_341/points" />
  <arg name="odometry_topic_arg" default="avt_341/odometry" />

  <node name="mavs_avt_vehcicle_node" pkg="mavs_avt_example" type="mavs_avt_vehicle_node" output="screen" required="true" >
    <remap from="mavs_avt/cmd_vel" to="$(arg cmd_vel_topic_arg)"/> 
    <remap from="mavs_avt/odometry" to="$(arg odometry_topic_arg)"/> 
    <param name="rp3d_vehicle_file" value="$(arg vehicle_file_arg)" />
    <param name= "debug_camera" value="true" />
    <param name = "Initial_X_Position" value = "-50.0"/>
    <param name = "Initial_Y_Position" value = "0.0"/>
    <param name = "Initial_Heading" value = "0.0"/>
    <param name="scene_file" value="$(arg scene_file_arg)"/>
    <param name = "soil_strength" value = "250"/> 
    <param name = "surface_type" value = "dry"/>  
    <param name= "publish_imu" value="true" />
    <param name = "imu_input_file" value = "$(find mavs_avt_example)/config/mavs_imu_inputs.json" />
    <param name= "use_human_driver" value="true" />
  </node>

  <node name="mavs_avt_sensors_node" pkg="mavs_avt_example" type="mavs_avt_sensors_node" output="screen" required="true" >
    <remap from="mavs_avt/point_cloud2" to="$(arg point_cloud_topic_arg)"/> 
    <param name= "use_lidar" value="true" />
    <param name= "display_lidar" value="true" />
    <param name= "register_lidar" value="true" />
    <param name="rp3d_vehicle_file" value="$(arg vehicle_file_arg)" />
    <param name = "lidar_type" value = "OS1"/>
    <param name="blanking_dist" value="5.0" />
    <param name="scene_file" value="$(arg scene_file_arg)"/>
  </node>


</launch>